### PR TITLE
WIP: Data validation before creation

### PR DESCRIPTION
**Pull request by trimailov**
WIP for data validation, for simple flat case (no nesting).

WIP, because not finished on validation implementation and tests.

### CODE REVIEW

File: spinta/types/model.py [hunk 1]
```diff
@@ -69,6 +69,15 @@ def check(context: Context, model: Model):
         context.deprecation("`id` property must specify real type like 'string' or 'integer'. Use of 'pk' is deprecated.")
 
 
+@check.register()
+def check(context: Context, model: Model, data: dict):
+    for name, prop in model.properties.items():
+        if name in data:
+            data_value = data[name]
+            if not prop.type.is_valid(data_value):
+                context.error(f"{data_value} is not valid type: {prop.type}")
```
Comment (sirex) [inline spinta/types/model.py]: `context.error` is an old leftover, don't use it. Probably here it is better to raise `ValidationError` or something like that, then you should handle these check errors and return `400` response.

File: spinta/types/type.py [hunk 2]
```diff
@@ -16,27 +18,64 @@ class Type:
...
+            except ValueError:
+                return False
+            else:
+                return True
+
+
+
+class DateTime(Date):
+
+    DEFAULT_FMT = "%Y-%m-%d %H:%M:%S"
```
Comment (sirex) [inline spinta/types/type.py]: Maybe use `fromisoformat`?

File: spinta/types/type.py [hunk 3]
```diff
@@ -16,27 +18,64 @@ class Type:
...
 
 class Integer(Type):
-    pass
+
+    def is_valid(self, value):
+        if isinstance(value, int):
+            return True
+        else:
+            try:
+                int(value)
```
Comment (sirex) [inline spinta/types/type.py]: Maybe add two step validation, using `load` and `check`? `load` would make sure, that value is loaded from a primitive value to a python native value, and then `check` would assume, that all types are already correct, and you only need to check if value is correct.

You will have to convert values to python-native values anyway.

Also a third step could be needed `prepare`, before saving data to database, python native values must be converted to a database native types. But maybe that is for another pull request.

Comment (trimailov) [reply spinta/types/type.py]: Do you mean to have something like:

```
data = load(context, model, data)
check(context,model, data)
check(context, model, model.backend, data)
```

in here: https://github.com/atviriduomenys/spinta/pull/19/files#diff-11540b6ded55485b62d4f20cf795971cR95

Comment (sirex) [reply spinta/types/type.py]: Yes, something like that. First data should be `load`ed, going down all the tree and then loaded data should be passed to `check`.

Comment (trimailov) [reply spinta/types/type.py]: Will refactor this with next PR.

Will merge as is (tests are passing).

File: spinta/types/type.py (hunk 2 above)
Comment (trimailov) [reply spinta/types/type.py]: Will do on next PR.

**Pull request merged**